esmvalcore/preprocessor/_regrid.py

```diff
@@ -185,8 +189,45 @@ def _attempt_irregular_regridding(cube, scheme):
     return False
 
 
+def _add_regridded_aux_coords(cube, regridded_cube, target_grid):
+    """Get regridded :class:`iris.coords.AuxCoord` and corresponding dims."""
+    if regridded_cube.ndim != cube.ndim:
+        return
+    for coord in cube.coords(dim_coords=False):
+        if regridded_cube.coords(coord.name()):
+            continue
+        if coord.ndim < 2:
+            continue
+        if coord.bounds is not None:
+            logger.warning(
+                "Auxiliary coordinate '%s' cannot be regridded since it "
+                "contains bounds", coord.name())
+            continue
+        logger.debug(
+            "Regridding (bound-less) auxiliary coordinate '%s' which could "
+            "not be regridded by iris", coord.name())
+
+        # Broadcast to shape of whole cube
+        coord_dims = cube.coord_dims(coord)
+        points = iris.util.broadcast_to_shape(coord.core_points(),
+                                              cube.shape,
+                                              coord_dims)
+        aux_coord_cube = cube.copy(points)
+        aux_coord_cube = aux_coord_cube.regrid(
+            target_grid, Linear(extrapolation_mode='extrapolate'))
```

Comment:
any reason why we fix the extrapolation method?

Reply:
Do other methods make sense for coordinates? Nearest neighbor? Area weighting?

Reply:
I dunno, preserving the original functional argument `HORIZONTAL_SCHEMES[scheme]` seems a bit more general, if stuff hits the fan we can always go back and restrict

Reply:
I think that scientifically only `linear` makes sense, but I'm not 100% sure. Consider this example:

Original grid (`lat x lon`): `[1.0, 2.0] x [1.0, 2.0]`
Target grid: `[1.25, 1.75] x [1.25, 1.75]`

If we have an 2D auxiliary coordinate that is basically identical to `lat` coord, i.e.

```
[[1.0, 1.0], [2.0, 2.0]]
```

I except that this is regridded to

```
[[1.25, 1.25], [1.75, 1.75]]
```

That is only true for the `linear` regridding.

Reply:
what are the other methods giving?

Reply:
my worry is not necessarily a technical one but rather - if one method is applied to DimCoords and another method to AuxCoords that is mixing statistics in one operation, bit of a black box for the user and that might bite at some point and nobody will know why (all silent). If so then I'd add a logger message saying `Hell man, you might in trouble, am changing your regridding method...`

Reply:
`nearest` gives

```
[[1.0, 1.0], [2.0, 2.0]]
```

`area_weighted` gives

```
[[--, --], [--, 2.0]]
```

Reply:
Gah, pretty funky! OK keep the fixed one then, clearly, but I'd the message still pls, we should be as transparent as possible :tumbler_glass:

Reply:
> my worry is not necessarily a technical one but rather - if one method is applied to DimCoords and another method to AuxCoords that is mixing statistics in one operation, bit of a black box for the user and that might bite at some point and nobody will know why (all silent). If so then I'd add a logger message saying `Hell man, you might get in trouble, am changing your regridding method...`

But the `DimCoords` are technically not regridded; they rather define what the target grid looks like, so they are fixed. Like I said, I'm not 100% sure if the regridding of other aux coords makes scientifically sense in the first place (also since `iris` does obviously not support it), but my intuition tells me to use a linear scheme. Maybe @zklaus can comment on this since he's done the irregular regridding?

Reply:
I think doing it linearly is indeed the only thing that might make some sense, so I think this conversation can be resolved. I'll but general comments below.

Reply:
Ok let's but this converstion :grin: